Scripts/application.py

```diff
@@ -0,0 +1,112 @@
+import sqlite3
+import json
+from typing import Dict
+from flask import Flask, request
+app = Flask(__name__)
+
+path_to_db = '../db/allYourSantaAreBelongToUs.db'
+
+@app.route('/')
+def hello_world():
+    return 'Hello World!'
+    
+@app.route('/api/v1/person', methods = ["GET","POST"])
+def return_all_people():
+    
+    if request.method == "POST":
+        posted_email = request.get_json()['email']
+        with sqlite3.connect(path_to_db) as connection:
+            cursor = connection.cursor()
+            cursor.execute('INSERT into people (email) VALUES (?);', (posted_email,))            
+    
+    # returns a json of all people
+    with sqlite3.connect(path_to_db) as connection:
+        cursor = connection.cursor()
+        #get all of the people
+        cursor.execute('SELECT person_ID, email, givingTo_ID FROM people')
+        raw_data: List = cursor.fetchall()
+        data: Dict = {}
+        for person in raw_data:
+            data[person[0]] = {"email": person[1], "assignment ID": person[2]}
+        json_data = json.dumps(data)
+    return json_data
+    
+@app.route('/api/v1/person/<person_id>', methods = ["GET","DELETE"])
+def return_person(person_id: str) -> str:
+    if request.method == "DELETE":
+        with sqlite3.connect(path_to_db) as connection: 
+            cursor = connection.cursor()
+            try:
+                command = 'DELETE FROM people WHERE person_ID=?;'
+                cursor.execute(command, (person_id,))
+                command = 'DELETE FROM forbiddenPairings WHERE santa_ID=? or assignment_ID=?;'
+                cursor.execute(command, (person_id, person_id))
+            except:
+                return('There was an error. Check that the requested ID is valid.')
+        return return_all_people()
+    
+    if request.method == "GET":
+        # returns a json of <person_id>
+        with sqlite3.connect(path_to_db) as connection: 
+            cursor = connection.cursor()
+            try:
+                command = 'SELECT person_ID, email, givingTo_ID FROM people WHERE person_ID=?;'
+                cursor.execute(command, (person_id,))
+                raw_data: List = cursor.fetchall()
+                data: Dict = {raw_data[0][0]:{"email": raw_data[0][1], "assignment ID": raw_data[0][2]}}
+                json_data = json.dumps(data)
+            except:
+                return('There was an error. Check that the requested ID is valid.')
+        return json_data
+    
+@app.route('/api/v1/forbidden_pairing', methods = ["GET","POST"])
+def return_forbidden_pairings():
+    
+    if request.method == "POST":
+        new_giver_constraint = request.get_json()["santa"]
+        new_receiver_constraint = request.get_json()["assignment"]

```

Comment:
oh no! have the caller use ids not emails. ids are always better.

Reply:
ok